internal/expr/parser.go

```diff
@@ -392,47 +426,36 @@ func (p *Parser) parseColumn() (columnExpr, bool, error) {
 	return fullName{}, false, nil
 }
 
-func (p *Parser) parseFuncName() (columnExpr, bool, error) {
+func (p *Parser) parseFunc() (columnExpr, bool, error) {
 	cp := p.save()
-	if p.skipName() && p.skipByte('(') {
-		bracketLevel := 1
+	scp := p.startSubpart()
+	if p.skipName() && p.peekByte('(') {
 		for {
-			if p.pos == len(p.input) {
-				return funcExpr{}, false, nil
-			}
-
-			if ok, err := p.skipStringLiteral(); err != nil {
+			if err := p.advance(); err != nil {
+				cp.restore()
 				return funcExpr{}, false, err
-			} else if ok {
-				continue
 			}
-			if ok := p.skipComment(); ok {
-				continue
+
+			if p.bracketLevel == 0 {
+				break
 			}
 
-			if p.skipByte('(') {
-				// Keep track of bracket nesting level in function to identify
-				// closing bracket.
-				bracketLevel++
-				continue
+			if p.pos == len(p.input) {
```

Comment:
Should these two break conditions be merged?

Reply:
Merged.